twilight-http/src/request/guild/auto_moderation/create_auto_moderation_rule.rs

```diff
@@ -229,51 +244,104 @@ impl<'a> CreateAutoModerationRule<'a> {
+            validate_auto_moderation_exempt_roles(exempt_roles)?;
             fields.exempt_roles = Some(exempt_roles);
 
-            fields
+            Ok(fields)
         });
 
         self
     }
 
     /// Create the request with the trigger type [`Keyword`], then execute it.
     ///
-    /// Rules of this type require the `keyword_filter` field specified, and
-    /// this method ensures this. See [Discord Docs/Keyword Matching Strategies]
-    /// and [Discord Docs/Trigger Metadata].
+    /// Rules of this type require the `keyword_filter`, `regex_patterns` and
+    /// `allow_list` fields specified, and this method ensures this.
+    /// See [Discord Docs/Keyword Matching Strategies] and
+    /// [Discord Docs/Trigger Metadata] for more information.
+    ///
+    /// Only rust-flavored regex is currently supported by Discord.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`AutoModerationMetadataKeywordFilter`] if the `keyword_filter`
+    /// field is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataKeywordFilterItem`] if a `keyword_filter`
+    /// item is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataAllowList`] if the `allow_list` field is
+    /// invalid.
+    ///
+    /// Returns [`AutoModerationMetadataAllowListItem`] if an `allow_list` item
+    /// is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataRegexPatterns`] if the `regex_patterns`
+    /// field is invalid.
+    ///
+    /// Returns [`AutoModerationMetadataRegexPatternsItem`] if a `regex_patterns`
+    /// item is invalid.
     ///
     /// [`Keyword`]: AutoModerationTriggerType::Keyword
     /// [Discord Docs/Keyword Matching Strategies]: https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies
     /// [Discord Docs/Trigger Metadata]: https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-trigger-metadata
+    /// [`AutoModerationMetadataKeywordFilter`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataKeywordFilter
+    /// [`AutoModerationMetadataKeywordFilterItem`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataKeywordFilterItem
+    /// [`AutoModerationMetadataAllowList`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataAllowList
+    /// [`AutoModerationMetadataAllowListItem`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataAllowListItem
+    /// [`AutoModerationMetadataRegexPatterns`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataRegexPatterns
+    /// [`AutoModerationMetadataRegexPatternsItem`]: twilight_validate::request::ValidationErrorType::AutoModerationMetadataRegexPatternsItem
     pub fn with_keyword(
         mut self,
         keyword_filter: &'a [&'a str],
+        regex_patterns: &'a [&'a str],
+        allow_list: &'a [&'a str],
     ) -> ResponseFuture<AutoModerationRule> {
```

Comment:
implementors of `TryIntoRequest` having methods that call `into_future()` for you isn't a common practice in twilight, instead, do we want to take an enum of these parameters in the method on `Client`, similar to `InteractionClient::create_response`

although this isn't something the pr changes, we can include it here

```rs
client.create_auto_moderation_rule(
    Id::new(123),
    "1984",
    AutoModerationEventType::MessageSend,
    AutoModerationTrigger::Keyword(AutoModerationTriggerKeyword {
        keyword_filter: &["js"],
        regex_patterns: &["/./"],
        allow_list: &[],
   })
)
```